File: top_interview_quest/easy/remove_duplicates_sorted_array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int siz = nums.size();
        int ind = 1; 
        for (int i=1;i<siz;i++) {
                if (nums[i]!=nums[i-1]) {
                    nums[ind] = nums[i];
                    ind++;
                }
            } 
        return ind; }
};


//alternate
class Solution 
{
public:
    int removeDuplicates(vector<int>& nums, int k = 1)
    {
		return distance(nums.begin(), unique(nums.begin(), nums.end()));
		// or just subtract iterators...
		// return unique(nums.begin(), nums.end()) - nums.begin();
    }
};

File: top_interview_quest/easy/remove_duplicates_sorted_array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        j = 0
        k=0
        l = len(nums)
        while k!=len(nums):
          if nums[j] in nums[:j]:
              temp = nums[j]
              nums.pop(j)
              nums.append(temp)
          else:
              j+=1
          k+=1
        return j



##another s=way

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        size = len(nums)
        insertIndex = 1
        for i in range(1, size):
            # Found unique element
            if nums[i - 1] != nums[i]:      
                # Updating insertIndex in our main array
                nums[insertIndex] = nums[i] 
                # Incrementing insertIndex count by 1 
                insertIndex = insertIndex + 1       
        return insertIndex